test/FeeCollector.test.sol

```diff
@@ -0,0 +1,737 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import {Test, console2} from "forge-std/Test.sol";
+import {ERC1967Proxy} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import {FeeCollector, IReceiver, ICoreContract} from "../src/FeeCollector.sol"; // adjust the path if needed
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+/// Minimal ERC20 token used as fee token (maxBTC)
+contract MockERC20 {
```

Comment:
You could have inherited a lot of functionality in this mock from the standard openzeppelin ERC20 implementation, which means:
- less code
- less potential for errors in tests